taskgraph/Task.py

```diff
@@ -1002,27 +1051,31 @@ def _call(self):
                                             artifact_target,
                                             result_target_path_stats,
                                             self._target_path_list))
-                            result_calculated = True
+                            artifact_copied = True
             except IOError as e:
                 LOGGER.warning(
                     "IOError encountered when hashing original source "
                     "files.\n%s" % e)
-        if not result_calculated:
+        if not artifact_copied:
             if self._worker_pool is not None:
                 result = self._worker_pool.apply_async(
                     func=self._func, args=self._args, kwds=self._kwargs)
                 # the following blocks and raises an exception if result
                 # raised an exception
                 LOGGER.debug("apply_async for task %s", self.task_name)
-                result.get()
+                self._result = result.get()
             else:
                 LOGGER.debug("direct _func for task %s", self.task_name)
-                self._func(*self._args, **self._kwargs)
+                self._result = self._func(*self._args, **self._kwargs)
 
         # check that the target paths exist and record stats for later
+        if not self._hash_target_files:
+            target_hash_algorithm = 'exists'
+        else:
+            target_hash_algorithm = self._hash_algorithm

```

Comment:
I get what you're saying, but I think this is still okay. I specifically want to differentiate between the "hash algorithm" and the "target file hash algorithm". Usually the "hash algorithm" is fine for all the files and if it's not it's only the case when the target files need to exist. 

In turn `_get_file_stats` shouldn't care if the "file" is input or output, just what it should do. So that's why I've got that intermediate step in this bit of code to differentiate what the `target_hash_algorithm` is on the one case when we only care if the files exist.

Reply:
Yep, makes sense why this will be useful, and thanks for clarifying!